### DISCUSSION TITLE
External (GEM) should there be Help and Reference files

### DISCUSSION THREAD

**Discussion by kazakore** [plugdata-team/plugdata#1454] [Q&A]
I've added the GEM external as I wanted to try and start making a MIDI to GFX patchc for a project. At first I didn't think it had loaded as it's not listed at startup but searching and inserting an object works and it then appears to load everything.

But I am very new to it and going to the Help and Reference pages brings up blanks. Do these exist for GEM? And if so is there a way to get them showing in PlugData?

I tried adding GEM as a library loaded at startup, rather than just being a search path and it throws up this error.

```Failed to load library: /...../Gem/```  (with full path.)

I don't know if that's normal...

**Comment by dromer** [comment]
It's here for me:
![](https://i.imgur.com/sTeeLyR.png)

**Reply by dromer** [reply]
This is of course with the latest develop builds, not any released version.

**Comment by kazakore** [comment]
Thanks. Didn't think to look there. Any way to get it working on the right-click Help option? At least this will allow me to get started though :)

**Comment by kazakore** [comment]
OK I see what happened!!!

I'm *not* in fact getting the GEM Objects! I'm only getting the Gem-Help Objects when I start typing an object name from GEM.

**Comment by kazakore** [comment]
![Screenshot_2024-02-24_11-56-20](https://github.com/plugdata-team/plugdata/assets/8245916/dcf6639c-05db-42a7-9193-3a0ad952fd9d)

**Comment by kazakore** [comment]
OK I can load a GEM Object, I have to start searching for it, find the version that ends -help and then delete the -help part of the text and hit ESC to get it to create it with the "partial" text, then it seems to be there. (I do have a GLX error increasing by the second and no idea if it's actually working at the moment.)

Why is the Search autocomplete completing it for the Help files instead of the actual Objects themselves?

Also when I do view Help files their formatting and theme is completely different and the examples contained within them can not be copy and pasted to the main page like with other Help pages. Is this expected from Externals or should they be the same as the default ones?

**Reply by dromer** [reply]
Did you still modify your startup config?

**Reply by kazakore** [reply]
I added it through the internal Find Externals menu. After adding I click Add Search button and I've tried with and without ~/.../externals/Gem in the Load Library At Startup and it behaves the same way (with the Failed to load library error mentioned above if I have the path set.) 

Is there something else I should do? Sorry if I missed something, last time I looked at Vanilla was about 10 years ago and used pd-extended so everything was just there.

**Comment by 60-hz** [comment]
- Check the error in the console saying gem fail to load
- If gem is not loaded, you won't see the object in autocomplete, only help...
- The Gem found in deken is not  compatible with Plugdata...
- PlugData 0.8.3 don't have Gem. The only way to load Gem is to download latest nightly Plugdata version from https://plugdata.org/download.html

**Comment by kazakore** [comment]
Sorry I really should have provided OS and Versions in my initial post!

Running on Manjaro, installed plugdata-bin from AUR, which I thought should be official release and not the nightly/betas (that would usually be the plugdata-git option that was also there.) ```aur/plugdata-bin 0.8.3-1 (+2 0.00) (Installed)``` is what my system shows installed and the screenshot shows Gem as listed as available (and installed) External.


Gem is there in the Find Externals menu on this version and that's how I've tried to install it. Running the Examples such as 02.blueRectangle works (to an extent*) but even after loading and running an Example autocomplete only shows Help files.

(*I definitely have other bugs with Gem I need to resolve anyway. It crashes after a while and maybe you can see the error message in the screenshot which is counting up quite rapidly. This side of things I don't so much expect help from you, unless I'm lucky and it's an obvious issue.)


![Screenshot_2024-02-24_14-56-29](https://github.com/plugdata-team/plugdata/assets/8245916/30cc5e5b-1c3b-4b66-9263-97639c260d66)
